src/storage/recovery/recovery_manager.cpp

```diff
@@ -104,24 +113,14 @@ void RecoveryManager::RecoverFromLogs(const common::ManagedPointer<AbstractLogPr
     }
     buffered_changes_map_.clear();
   }
-
-  if (replication_manager_ != DISABLED &&
-      log_provider->GetType() == AbstractLogProvider::LogProviderType::REPLICATION) {
-    auto rep_log_provider = log_provider.CastManagedPointerTo<ReplicationLogProvider>();
-    rep_log_provider->LatchPrimaryAckables();
-    std::vector<uint64_t> &ackables = rep_log_provider->GetPrimaryAckables();
-    std::vector<uint64_t> ackables_copy = ackables;
-    ackables.clear();
-    rep_log_provider->UnlatchPrimaryAckables();
-    for (const uint64_t primary_cb_id : ackables_copy) {
-      replication_manager_->ReplicaAck("primary", primary_cb_id, false);
-    }
-  }
 }
 
 void RecoveryManager::ProcessCommittedTransaction(noisepage::transaction::timestamp_t txn_id) {
   // Begin a txn to replay changes with.
   auto *txn = txn_manager_->BeginTransaction();
+  if (replication_manager_ != DISABLED && replication_manager_->IsReplica()) {
+    txn->SetReplicationPolicy(transaction::ReplicationPolicy::DISABLE);
```

Comment:
Might be more efficient to just set the default replication policy to `DISABLE` in DB main for replica nodes.

Reply:
The semantics of that are slightly different I think (disable replication for all txns vs disable replication for just txns that we are replaying via replication). We definitely want the latter, I'm not sure that we want the former further down the road (if we ever support multi-master, big doubts though).

Reply:
Well, by a similar token you can argue that those txns which should be replayed can request it explicitly. And you're right that it will save the if check. So I'm moving it anyway.

Reply:
done